src/sources/dnstap/mod.rs

```diff
@@ -50,73 +55,67 @@ pub struct DnstapConfig {
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
     pub host_key: Option<OptionalValuePath>,
 
-    /// Absolute path to the socket file to read DNSTAP data from.
-    ///
-    /// The DNS server must be configured to send its DNSTAP data to this socket file. The socket file is created
-    /// if it doesn't already exist when the source first starts.
-    pub socket_path: PathBuf,
-
     /// Whether or not to skip parsing or decoding of DNSTAP frames.
     ///
     /// If set to `true`, frames are not parsed or decoded. The raw frame data is set as a field on the event
     /// (called `rawData`) and encoded as a base64 string.
-    raw_data_only: Option<bool>,
+    pub raw_data_only: Option<bool>,
 
     /// Whether or not to concurrently process DNSTAP frames.
     pub multithreaded: Option<bool>,
 
     /// Maximum number of frames that can be processed concurrently.
     pub max_frame_handling_tasks: Option<u32>,
 
-    /// Unix file mode bits to be applied to the unix socket file as its designated file permissions.
-    ///
-    /// Note: The file mode value can be specified in any numeric format supported by your configuration
-    /// language, but it is most intuitive to use an octal number.
-    pub socket_file_mode: Option<u32>,
-
-    /// The size, in bytes, of the receive buffer used for the socket.
-    ///
-    /// This should not typically needed to be changed.
-    #[configurable(metadata(docs::type_unit = "bytes"))]
-    pub socket_receive_buffer_size: Option<usize>,
-
-    /// The size, in bytes, of the send buffer used for the socket.
-    ///
-    /// This should not typically needed to be changed.
-    #[configurable(metadata(docs::type_unit = "bytes"))]
-    pub socket_send_buffer_size: Option<usize>,
-
     /// The namespace to use for logs. This overrides the global settings.
     #[configurable(metadata(docs::hidden))]
     #[serde(default)]
-    log_namespace: Option<bool>,
+    pub log_namespace: Option<bool>,
 }
 
 fn default_max_frame_length() -> usize {
     bytesize::kib(100u64) as usize
 }
 
+/// Listening mode for the `dnstap` source.
+#[configurable_component]
+#[derive(Clone, Debug)]
+#[serde(untagged)]
+#[configurable(metadata(docs::enum_tag_description = "The type of dnstap socket to use."))]
+#[allow(clippy::large_enum_variant)] // just used for configuration
```

Comment:
```
#[serde(tag = "mode", rename_all = "snake_case")]
```

Can we add this to keep things consistent with the socket source. So to configure  the mode, we will need:

```yml
mode: tcp
```

This will be a breaking change, but a small one and I think worth it for the sake of being consistent.

Reply:
Makes sense. I tried to avoid making a breaking change, but it isn't that bad really.

Reply:
This will need an associated note in the changelog to say something like:

```
As the `dnstap` source now supports multiple socket types, you will need to update your configuration to specify which type - either `mode: unix` for the existing unix sockets mode or `mode: tcp` for the new tcp mode.
```

Reply:
Does it need a special tag to mark it as breaking?

Reply:
Yeah, if you rename the file to `changelog.d/19892_dnstap_over_tcp.breaking.md` that should do it.

Can you also rename this PR to add a `!` so:

`feat(sources)!: add TCP mode to DNSTAP source`

Thanks